codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -77,6 +84,17 @@ public DefaultHttp2ConnectionDecoder(Http2Connection connection,
         connection.local().flowController().frameWriter(encoder.frameWriter());
     }
 
+    public DefaultHttp2ConnectionDecoder(Http2Connection connection,
+                                         Http2ConnectionEncoder encoder,
+                                         Http2FrameReader frameReader,
+                                         Http2Settings initialSettings,
+                                         Http2PromisedRequestVerifier requestVerifier) throws Http2Exception {
+        this(connection, encoder, frameReader, requestVerifier);
+        if (initialSettings != null) {
+            applyLocalSettings(initialSettings);
```

Comment:
@Scottmitch 

> The responsibility for retaining the settings values and enforcing the settings constraints is spread out in different areas of the code and may be initialized with different values than the default specified in the RFC.

Yes, this is true ... I think we need to make sure that our defaults conform to the RFC. They must not rely on the initial settings frame to be applied before they are spec-compliant.

> It can be difficult to know how to configure all the places where the settings are enforced, so this Http2Settings object allows the initial settings to be configured consistently in a single place, and ensure the initial internal settings are also communicated to the peer.

Ok, I think I understand what you're trying to do now :).  However, I think this PR is conflating two concepts that need to be kept separate:

1. The `default` settings identified in the spec
2. The `initial settings` specified by the endpoint.

Default settings are essentially a constant - this is how we should go through all of our objects and set everything up. This can be done a couple of ways: as I replied above, we could just make the defaults in all of the various objects spec-compliant ... or, we could make a `default settings` object that we apply in the decoder as part of the construction process.

We can't immediately apply `initial settings` however, since they are specified by the endpoint (i.e. the application). For example, we can't just shrink the allowed frame size immediately because the remote endpoint will almost certainly violate it.  Instead, we have to wait for the remote endpoint to confirm receipt of the settings.

Does that make sense or am I missing something?